File: nim/fileio.nim
import os, strutils

const
  ReadBufferSize = 65536
  ChunkSize = 65536

proc main() =
  let fileCount = paramStr(1).parseInt()
  let n = paramStr(2).parseInt()

  # Pre-create content with n '1' characters
  var content = newString(n)
  for i in 0..<n:
    content[i] = '1'

  # Write phase
  for i in 0..<fileCount:
    let filename = "file" & $i
    writeFile(filename, content)

  # Read phase - read in chunks and count total bytes
  var totalBytes: uint64 = 0
  var buffer = newString(ReadBufferSize)

  for i in 0..<fileCount:
    let filename = "file" & $i
    let file = open(filename, fmRead)

    while true:
      let bytesRead = file.readBuffer(addr buffer[0], ReadBufferSize)
      if bytesRead == 0:
        break
      totalBytes += uint64(bytesRead)

    file.close()

  # Count chunks of 65536 bytes
  let totalChunks = totalBytes div ChunkSize
  echo totalChunks

  # Cleanup
  for i in 0..<fileCount:
    let filename = "file" & $i
    removeFile(filename)

main()

File: perl/fileio.pl
#!/usr/bin/perl
use strict;
use warnings;

my $file_count = $ARGV[0];
my $n = $ARGV[1];

my $READ_BUFFER_SIZE = 65536;
my $CHUNK_SIZE = 65536;

# Pre-create content with n '1' characters
my $content = "1" x $n;

# Write phase
for my $i (0..$file_count-1) {
    my $filename = "file$i";
    open my $fh, '>', $filename or die;
    binmode $fh;
    print $fh $content;
    close $fh;
}

# Read phase - read in chunks and count total bytes
my $total_bytes = 0;
my $buffer;
for my $i (0..$file_count-1) {
    my $filename = "file$i";
    open my $fh, '<', $filename or die;
    binmode $fh;
    while (read($fh, $buffer, $READ_BUFFER_SIZE)) {
        $total_bytes += length($buffer);
    }
    close $fh;
}

# Count chunks of 65536 bytes
my $total_chunks = int($total_bytes / $CHUNK_SIZE);
print "$total_chunks\n";

# Cleanup
for my $i (0..$file_count-1) {
    my $filename = "file$i";
    unlink $filename;
}